src/Microsoft.ComponentDetection.Detectors/conda/CondaDependencyResolver.cs

```diff
@@ -0,0 +1,110 @@
+namespace Microsoft.ComponentDetection.Detectors.Poetry;
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.ComponentDetection.Detectors.Poetry.Contracts;
+
+public static class CondaDependencyResolver
+{
+    /// <summary>
+    /// Iterate through all dependencies that are explicitly referenced in the conda environment
+    /// and build as well as register the full dependency tree for each of them.
+    /// </summary>
+    /// <param name="condaLock">The full condaLock object.</param>
+    /// <param name="explicitDependencies">The list of all dependencies that are explicitly specified in the conda environment.</param>
+    /// <param name="singleFileComponentRecorder">The SingleFileComponentRecorder.</param>
+    public static void RecordDependencyGraphFromFile(CondaLock condaLock, List<string> explicitDependencies, ISingleFileComponentRecorder singleFileComponentRecorder)
+    {
+        if (condaLock != null && condaLock.Package != null)
```

Comment:
nit: equivalent expression
```suggestion
        if (condaLock?.Package != null)
```

Reply:
Great call out! Adapted.